src/main/python/variants_pca.py

```diff
@@ -0,0 +1,200 @@
+        row_count: The size (N) of the N x N matrix.
+
+    Returns:
+        An RDD[int, (int, float)] representing centered rows. The first int is
+        the row index, the (int, float) tuple is the column index, and the
+        centered value.
+    """
+
+    # Row-by-row (row major) RDD. Each row is a list of (column, value).
+    entries = (sim_matrix.
+        map(lambda ((y, x), v): (y, (x, float(v)))).
+        groupByKey().
+        sortByKey(True).
+        cache()
+    )
+    row_sums = entries.map(lambda (y, xvs): sum(v for (x, v) in xvs)).collect()
+    matrix_sum = sum(row_sums)
+    matrix_mean = float(matrix_sum) / row_count / row_count
+
+    sc = pyspark.SparkContext._active_spark_context
+    broadcast_row_sums = sc.broadcast(row_sums)
+
+    def center_rows((row, col_vals)):
+        row_mean = broadcast_row_sums.value[row] / float(row_count)
+
+        def center_cols(col, val):
+            col_mean = broadcast_row_sums.value[col] / float(row_count)
+            return (col, val - row_mean - col_mean + matrix_mean)
+
+        return [center_cols(col, val) for col, val in col_vals]
+
+    return entries.map(center_rows)
+
+def perform_pca(matrix, row_count, nr_principal_components=2):
+    """Return principal components of the input matrix.
+
+    This function uses MLlib's ``RowMatrix`` to compute principal components.
+
+    Args:
+        matrix: An RDD[int, (int, float)] representing a sparse matrix. This
+            is returned by ``center_matrix`` but it is not required to center
+            the matrix first.
+        row_count: The size (N) of the N x N ``matrix``.
+        nr_principal_components: Number of components we want to obtain. This
+            value must be less than or equal to the number of rows in the input
+            square matrix.
+
+    Returns:
+        An array of ``nr_principal_components`` columns, and same number of rows
+        as the input ``matrix``. This array is a ``numpy`` array.
+    """
+
+    py_rdd = matrix.map(lambda row: linalg.Vectors.sparse(row_count, row))
+    sc = pyspark.SparkContext._active_spark_context
+    java_rdd = mllib_common._py2java(sc, py_rdd)
+    scala_rdd = java_rdd.rdd()
+    sc = pyspark.SparkContext._active_spark_context
+    row_matrix = (sc._jvm.org.apache.spark.mllib.linalg.distributed.
+        RowMatrix(scala_rdd)
+    )
+    pca = row_matrix.computePrincipalComponents(nr_principal_components)
+    pca = mllib_common._java2py(sc, pca)
+    return pca.toArray()
+
+def pca(argv):
+    sc = pyspark.SparkContext._active_spark_context
+    args = sc._jvm.java.util.ArrayList()
+    for arg in argv:
+        args.append(arg)
+    args = sc._jvm.scala.collection.JavaConversions.asScalaBuffer(args)
+    jsc = sc._jsc.sc()
+
+    pca_conf = (sc._jvm.com.google.cloud.genomics.spark.examples.
+        PcaConf(args))
+    pca_driver = (sc._jvm.com.google.cloud.genomics.spark.examples.
+        VariantsPcaDriver(pca_conf, jsc))
```

Comment:
This can be done in a future PR but it would be cool to refactor what you are using from VariantsPcaDriver and not use VariantsPcaDriver directly.

Looks like you have quite a large portion of it in here already!